src/async_buffer_redirect.rs

```diff
@@ -0,0 +1,70 @@
+use gag::Redirect;
+use std::fs::File as StdFile;
+use std::io;
+use std::pin::Pin;
+use std::task::Context;
+use std::task::Poll;
+use tempfile::NamedTempFile;
+use tokio::fs::File;
+use tokio::io::AsyncRead;
+use tokio::io::ReadBuf;
+
+/// Buffer output in an in-memory buffer. Async version of [`gag::BufferRedirect`].
+pub struct AsyncBufferRedirect {
+    #[allow(dead_code)]
+    redir: Redirect<StdFile>,
+    outer: File,
+}
+
+/// An in-memory read-only buffer into which [`AsyncBufferRedirect`] buffers output. Async version
+/// of [`gag::Buffer`].
+pub struct AsyncBuffer(File);
+
+impl AsyncRead for AsyncBuffer {
+    #[inline(always)]
+    fn poll_read(
```

Comment:
I'm copying the inline attributes from the original `BufferRedirect` implementation.